OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -230,22 +232,15 @@ class CourseModalDialogFragment : DialogFragment() {
         }.show(childFragmentManager, null)
     }
 
-    private fun showUpgradeCompleteDialog() {
-        AlertDialogFragment.newInstance(
-            getString(R.string.title_upgrade_complete),
-            getString(R.string.upgrade_success_message),
-            getString(R.string.label_continue),
-            null,
-            null,
-            null
-        ).show(childFragmentManager, null)
-    }
-
     override fun onDestroyView() {
         super.onDestroyView()
         billingProcessor?.disconnect()
     }
 
+    fun setDialogDismissListener(listener: DialogDismissListener) {
+        this.dialogDismissListener = listener
+    }
+
```

Comment:
IMO, A part of the constructor is better here.

Reply:
removed